EchoPro/survey.py

```diff
@@ -36,741 +25,457 @@ class Survey:
+        Loads the biological, NASC, and stratification
+        data using parameters obtained from the configuration
+        files. This will generate data attributes associated with the tags
+        defined in both the configuration yml files and the reference CONFIG_MAP
+        and LAYER_NAME_MAP dictionaries.
+        """
+
+        ### Check whether data files defined from the configuration file exists
+        # Generate flat JSON table comprising all configuration parameter names
+        flat_configuration_table = pd.json_normalize(self.config)
+
+        # Parse the flattened configuration table to identify data file names and paths
+        parsed_filenames = flat_configuration_table.filter(regex="filename").values.flatten()
+
+        # Append the root directory to the parsed data file names
+        data_file_paths = [os.path.join(self.config['data_root_dir'], filename) for filename in parsed_filenames]
+
+        # Evaluate whether either file is missing
+        data_existence = [os.path.exists(file_path) for file_path in data_file_paths]
+
+        # Assign the existence status to each configuration file for error evaluation
+        data_status = dict(zip(data_file_paths, data_existence))
+
+        # Error evaluation and print message (if applicable)
+        if not all(data_status.values()):
+            missing_data = [data_file for data_file, exists in data_status.items() if not exists]
+            raise FileNotFoundError(f"The following data files do not exist: {missing_data}")
+        
+        ### Data validation and import
+        # Iterate through each file defined/detected in 'data_file_paths'
+        for file in data_file_paths:
+
+            # Parse data layers, associated configuration key tag, and the Excel sheet name associated
+            # with the data
+            data_layers, reference_path, sheet_name = self.search_nested_dict(self.config, file)
+
+            # Reference LAYER_NAME_MAP to update 'self.config' -- this is necessary for both mapping
+            # the data attribute structures and reorganizing the dictionary organization within each
+            # class attribute later on
+            # Pull LAYER_NAME_MAP reference information
+            layer_config = LAYER_NAME_MAP.get(data_layers[0])
+
+            # Determine how deep 'data_layers' is represented within the dictionary
+            # This is required for mapping dataframes with an appropriately named
+            # data layer name appended with '_df' (e.g. 'length' -> 'length_df')
+            data_name_bool = set(data_layers).intersection(layer_config['data'])
+            data_layer_index = data_layers.index(''.join(data_name_bool))
+
+            #pull_nested_dict(CONFIG_MAP, data_layers)
+            # Retrieve/access CONFIG_MAP values defining the expected column names and data types
+            validation_settings = pull_nested_dict(CONFIG_MAP, data_layers[:data_layer_index+1])
```

Comment:
The two mapping dict `LAYER_NAME_MAP` and `CONFIG_MAP` both have simple and known structures, and I think it is overkill to have dedicated functions like `search_nested_dict` and `pull_nested_dict` to find where some attributes are stored. These are essentially re-doing what XML tree was trying to achieve, and there are libraries that handles that already (note I am _not_ suggesting that you use XML! that would also be overkill).

I suggest that you just hard-coded the part about looking for `filename` and `sheetname`, given the known structures in the mappings. The code will be much easier to follow and debug/maintain later, since it makes use of the hierarchical structure in the yaml file.